SingularityUI/app/components/machines/usage/SlaveAggregates.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { PropTypes } from 'react';
+import CircularProgressbar from 'react-circular-progressbar';
+import Utils from '../../../utils';
+import { STAT_NAMES, HUNDREDTHS_PLACE } from '../Constants';
+
+const getPctSlaveUsage = (slaves, slaveUsages, usageCallback, resourceCallback) => {
```

Comment:
`Mapper` might be a more descriptive suffix than `Callback` for the parameter names. You could also avoid passing around the `slaves` and `slaveUsages` parameters around with
```
const getPctSlaveUsage = (usageMapper, resourceMapper) => (slaves, slaveUsages) => { //...}

const getCpuUtilizationPct = getPctSlaveUsage(
  usage => usage.cpusUsed,
  slave => Utils.getMaxAvailableResource(slave, STAT_NAMES.cpusUsedStat)
);

const getMemUtilizationPct = getPctSlaveUsage(
  usage => usage.memoryBytesUsed,
  slave => Utils.getMaxAvailableResource(slave, STAT_NAMES.memoryBytesUsedStat)
);
```